### PR TITLE
fixes dotnet/templating#2810: corrects source paths returned by GetCreationEffectsAsync method

### PR DESCRIPTION
**Pull request by vlada-shubina**
fixes dotnet/templating#2810: corrects source paths returned by GetCreationEffectsAsync method

The PR reverts changes done in https://github.com/dotnet/templating/pull/2760 as the fix broke the use case for the templates installed from nupkgs and applies different fix.

- adds the ability to pack the NuGet package to be used in IDE tests
- marked TestTemplates project as packable (non-shipping)
- adds FileRename tests to be run on nupkg additionally to templates installed from folder

### CODE REVIEW

**Review by DavidKarlas — COMMENTED** [review]
What happened to `RunnableProjectGenerator.cs`, that is not reviewable...
Should we make `IOrchestrator` obsolete, and tell API users to switch to `IOrchestrator2` and respect `IFileChange2.SourceRelativePath` in PostActions and other things?(when I say "tell" I mean inside `[Obsolete("message here")]`

**Comment by vlada-shubina** [comment]
> What happened to `RunnableProjectGenerator.cs`, that is not reviewable...
> Should we make `IOrchestrator` obsolete, and tell API users to switch to `IOrchestrator2` and respect `IFileChange2.SourceRelativePath` in PostActions and other things?(when I say "tell" I mean inside `[Obsolete("message here")]`

It seems I messed up the commit history... now last commit shows change to `RunnableProjectGenerator.cs`.
As for `IOrchestrator` - I actually reverted all changes I did previously, and kept fix to `RunnableProjectGenerator.cs`. I can make it obsolete now, but it became irrelevant to this fix.

**Review by bozturkMSFT — APPROVED** [review]
I have a few comments. It is hard to tell which parts were changed and which parts were just moved around. I hope I didn't miss anything important.

**Comment by vlada-shubina** [comment]
@bozturkMSFT changes that are relevant to fix are in the last commit, you can see and review it from here: https://github.com/dotnet/templating/pull/2827/commits/6b6ac6ecacd3d3cb00f24da3f70d447369479cb4

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs [hunk 1]
```diff
@@ -299,161 +243,179 @@ public bool TryGetTemplateFromConfigInfo(IFileSystemInfo templateFileConfig, out
...
+                    return false;
+                }
+                else
+                {
+                    bool boolVal = false;
+                    // Note: if the literal is ever null, it is probably due to a problem in TemplateCreator.Instantiate()
+                    // which takes care of making null bool -> true as appropriate.
+                    // This else can also happen if there is a value but it can't be converted.
+                    string val;
+                    while (environmentSettings.Host.OnParameterError(param, null, "ParameterValueNotSpecified", out val) && !bool.TryParse(val, out boolVal))
```
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs:278]: Nope. I could not make any sense of this code, but prefer not to touch it.

File: src/Microsoft.TemplateEngine.Core.Contracts/IOrchestrator.cs [hunk 2]
```diff
@@ -10,8 +10,19 @@ public interface IOrchestrator
...
+
+    public interface IOrchestrator2
+    {
+        void Run(string runSpecPath, IDirectory sourceDir, string targetDir);
+
+        void Run(IGlobalRunSpec spec, IDirectory sourceDir, string targetDir);
+
+        IReadOnlyList<IFileChange2> GetFileChanges(string runSpecPath, IDirectory sourceDir, string targetDir);
+
+        IReadOnlyList<IFileChange2> GetFileChanges(IGlobalRunSpec spec, IDirectory sourceDir, string targetDir);
```
Comment (bozturkMSFT) [inline src/Microsoft.TemplateEngine.Core.Contracts/IOrchestrator.cs:26]: So the previous commit removed this interface, but we are bringing it back? I was going to suggest default interface implementations again to not have the v2 of this interface (though I'm not sure how applicable to this case).

Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Core.Contracts/IOrchestrator.cs:26]: I've reverted all changes done in https://github.com/dotnet/templating/pull/2760 to original state in first commit, as fix was not complete.

**Review by DavidKarlas — APPROVED** [review]

**Pull request merged**